tools/rpc-client/RpcClient.hs

```diff
@@ -202,38 +244,206 @@ parsePostProcessing =
+                        Just err ->
+                            trace ("[Error] Request " <> r <> " failed: " <> BS.unpack err) $
+                                unless keepGoing $ do
+                                    shutdown skt ShutdownReceive
+                                    exitWith (ExitFailure 2)
+                        Nothing ->
+                            hPutStrLn stderr "[Info] Response matched with expected"
+                    pure mbError
+            shutdown skt ShutdownReceive
+            exitWith (if all isNothing results then ExitSuccess else ExitFailure 2)
+
+    -- complain on any errors in the tarball
+    throwAnyError :: Either Tar.FormatError Tar.FileNameError -> IO a
+    throwAnyError = either throwIO throwIO
+
+    -- unpack all rpc_*/*.json files into dir and return their names
+    unpackIfRpc :: FilePath -> Tar.Entry -> IO [FilePath] -> IO [FilePath]
+    unpackIfRpc tmpDir entry acc = do
+        case splitFileName (Tar.entryPath entry) of
+            -- assume single directory "rpc_<something>" containing "*.json" files
+            (_, "") -- skip all directories
+                | Tar.Directory <- Tar.entryContent entry ->
+                    acc
+            (dir, file) -- unpack json files into tmp directory
+                | "rpc_" `isPrefixOf` dir
+                , ".json" `isSuffixOf` file
+                , Tar.NormalFile bs _size <- Tar.entryContent entry -> do
+                    BS.writeFile (tmpDir </> file) bs
+                    (file :) <$> acc
+            _other ->
+                -- skip anything else
+                acc
+
+    noServerError :: IOException -> IO ()
+    noServerError e@IOError{ioe_type = NoSuchThing} = do
+        -- show instructions how to run the server
+        hPutStrLn stderr $ "[Error] Could not connect to RPC server on port " <> show port
+        hPutStrLn stderr $ "[Error] " <> show e
+        hPutStrLn stderr $
+            unlines
+                [ ""
+                , "To run the required RPC server, you need to"
+                , "1) extract `definition.kore` and `server_instance.json` from the tarball;"
+                , "2) look up the module name `<MODULE>` in `server_instance.json`;"
+                , "3) and then run the server using"
+                , "   $ kore-rpc definition.kore --main-module <MODULE> --server-port " <> show port
+                , ""
+                , "If you want to use `kore-rpc-booster, you should als compile an LLVM backend library"
+                , "by 1) extracting the `llvm_definition/` directory from the tarball;"
+                , "   2) running the llvm backend compiler on the unpacked files:"
+                , "      $ llvm-kompile llvm_definition/definition.kore llvm_definition/dt c`"
+                , "This will generate `interpreter.so` and you can run"
+                , "  `kore-rpc-booster definition.kore --main-module <MODULE> --llvm-backend-library interpreter.so`"
```

Comment:
this is great!!

Reply:
I second this: I was hot-testing these features on a bug report, and this was very helpful!